5.0/fr/0x19-V10-OAuth-and-OIDC.md

```diff
@@ -0,0 +1,169 @@
+
+## V10.1 Sécurité générique OAuth et OIDC
+
+Cette section couvre les exigences architecturales génériques qui s’appliquent à toutes les applications utilisant OAuth ou OIDC.
+
+| # | Description | Niveau |
+| :---: | :--- | :---: |
+| **10.1.1** | Vérifiez que les jetons sont envoyés uniquement aux composants qui en ont absolument besoin. Par exemple, lors de l'utilisation d'un modèle backend-for-frontend pour les applications JavaScript basées sur un navigateur, les jetons d'accès et d'actualisation ne doivent être accessibles qu'au backend. | 2 |
+| **10.1.2** | Vérifiez que le client n'accepte les valeurs du serveur d'autorisation (telles que le code d'autorisation ou le jeton d'identification) que si elles résultent d'un flux d'autorisation initié par la même session d'agent utilisateur et la même transaction. Cela nécessite que les secrets générés par le client, tels que la clé de preuve pour l'échange de codes (PKCE) « code_verifier », « state » ou le nonce OIDC, ne puissent être devinés, soient spécifiques à la transaction et soient liés de manière sécurisée au client et à la session d'agent utilisateur au cours de laquelle la transaction a été initiée. | 2 |
+
+## V10.2 Client OAuth
+
+Ces exigences détaillent les responsabilités des applications clientes OAuth. Le client peut être, par exemple, un serveur web back-end (souvent utilisé comme back-end pour front-end, BFF), une intégration de service back-end ou une application monopage front-end (SPA, également appelée application basée sur un navigateur).
+
+En général, les clients back-end sont considérés comme confidentiels et les clients front-end comme publics. Cependant, les applications natives exécutées sur l'appareil de l'utilisateur final peuvent être considérées comme confidentielles lors de l'enregistrement dynamique des clients OAuth.
+
+| # | Description | Niveau |
+| :---: | :--- | :---: |
+| **10.2.1** | Vérifiez que, si le flux de code est utilisé, le client OAuth dispose d'une protection contre les attaques de falsification de requêtes basées sur le navigateur, communément appelées falsification de requêtes intersites (CSRF), qui déclenchent des demandes de jetons, soit en utilisant la fonctionnalité de clé de preuve pour l'échange de code (PKCE), soit en vérifiant le paramètre « état » envoyé dans la demande d'autorisation. | 2 |
+| **10.2.2** | Vérifiez que, si le client OAuth peut interagir avec plusieurs serveurs d'autorisation, il dispose d'une protection contre les attaques par confusion. Par exemple, il peut exiger que le serveur d'autorisation renvoie la valeur du paramètre « iss » et la valide dans la réponse d'autorisation et la réponse du jeton. | 2 |
+| **10.2.3** | Vérifiez que le client OAuth demande uniquement les étendues requises (ou d’autres paramètres d’autorisation) dans les requêtes adressées au serveur d’autorisation. | 3 |
```

Comment:
"étendues" is not so clear. Shall we keep "scope"? (Alternatively, maybe "portée" would be a better translation than "étendues"?)

Edit : "portée" is used below for "scope".